webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFactServiceProxies.scala

```diff
@@ -74,12 +76,14 @@ import zio.syntax._
 class NodeFactInventorySaver(
     backend:               NodeFactRepository,
     val preCommitPipeline: Seq[PreCommit],
-    val basePostPipeline:  Seq[PostCommit[Unit]]
+    val basePostPipeline:  Seq[PostCommit[Unit]],
+    scoreServiceManager:   ScoreServiceManager
 ) extends PipelinedInventorySaver[Unit] {
 
   override def commitChange(inventory: Inventory): IOResult[Unit] = {
     implicit val cc = ChangeContext.newForRudder()
-    backend.updateInventory(FullInventory(inventory.node, Some(inventory.machine)), Some(inventory.applications)).unit
+    backend.updateInventory(FullInventory(inventory.node, Some(inventory.machine)), Some(inventory.applications)).unit *>
+    scoreServiceManager.handleEvent(InventoryScoreEvent(inventory.node.main.id, inventory)).unit
```

Comment:
Why puting it here and not in a `PostCommit` handler? They should be able to match that use case, and I don't see the need for hard implementation coupling here. And I don't think it gives any additional consistency benefits.

Reply:
done